### ISSUE TITLE
Flux test runner needs to understand the concept of a named root

### ISSUE THREAD

**Issue by nathanielc** [influxdata/flux#3566]
Currently the test runner does not know of a _root_ for the tests, it only understands relative paths to the current test file. This is a limitation based on our needs because we want to be able to extend tests from the Flux repo in downstream repos that consume Flux. 

Therefore we need the concept of a root so that tests can refer to other tests (via `extends`) based on their absolute path.

This will enable something like this to work:

```
// flux repo: stdlib/universe/last_test.flux
package universe_test

testcase last {
  // ....
}
```

```
// influxdb repo: query/stdlib/universe/last_test.flux
package universe_test

import "testing/expect"

testcase last_pushown extends "flux/stdlib/universe/last_test" {
    expect.planner(rules: ...)
    last_test.last()
}
```

Additionally tests within the same repo will be able to extend tests

```
// flux repo: stdlib/universe/last_other_test.flux
package universe_test

testcase last_other extends "flux/stdlib/universe/last_test" {
  last_test.last()
  // ... other stuff
}
```
## Implementation

How will this work? We will use a sentinel file to indicate where the root is where the contents of the file contain the name of the root.

The file name must be `fluxtest.root` the contents of the file are the name of the root (whitespace will be trimmed).

For example the `flux` repo will contain a file `fluxtest.root` at the root of the flux repo with `flux` as its contents. Similarly for the `influxdb` and other downstream repos.

This design allows for the test harness runners i.e `flux` and other CLI programs to work without explicit configuration other than which paths to include. This also means that local development is simple as the configuration is built into the repo filesystem instead of needing to be passed on as arguments to the CLI.

## Questions?

What should we name the file? `fluxtest.root` was the first thing that came to mind but there is likely a better name

Should we make the roots consistently the `stdlib` dir? That way the extends would read like 

```
testcase last_pushdown extends "flux/universe/last_test" { }
// or 
testcase last_pushdown extends "influxdb/universe/last_test" { }
```

**Comment by jsternberg** [comment]
The only thing I would change is `fluxtest.root` should be `flux.mod` and be a toml file with a single entry.

```
// flux.mod
module = "flux"
```

I have a feeling we're going to want this in the future and we might as well step onto a buildable base that can be discarded rather than something that will almost assuredly be discarded.

**Comment by rockstar** [comment]
Do we have a use where there are tests in influxdb that aren't `extends` tests? That was the confusion when talking about this issue: when would we ever want to extend a test that originates in influxdb?

Additionally, the syntax would likely have to be something like

    testcase last_pushdown extends flux/universe/last_test/test_name

...e.g. it would need to speak to the specific test. That might have been the intent here, but just wanted to call it out specifically.

**Comment by jsternberg** [comment]
@rockstar my main concern is I don't want to hardcode in any logic about where to find the tests. I don't think that we have to reference a specific test in the extends call. We just need a way to find the file.

**Comment by nathanielc** [comment]
@jsternberg Are test bundles going to be the same thing a Flux module? I am not sure and so I am hesitant to couple those ideas to start. Also I am not sure we want to use a `toml` file, the current spec for a module file is to make the file a Flux file. As both of these questions are not really relevant to our current problem I think we should try and avoid making any connection to future Flux modules. 

This is a simple enough change that we can update it once Flux modules are a more concrete design. In other words I fully expect to change this once Flux modules get built.

**Comment by nathanielc** [comment]
@rockstar To reference a specific test name you extend the file than call the function that corresponds to the test case you need.

```
testcase x extends y {
 y.test_name()
}
```
That is not unique to extending a test from influxdb. The same thing happens when extending a test from Flux.

**Comment by jsternberg** [comment]
Fixed by #3578.
